src/main/java/net/mcreator/ui/component/JStringListField.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2023, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.component;
+
+import net.mcreator.ui.dialogs.ListEditorDialog;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.init.UIRES;
+import net.mcreator.ui.validation.Validator;
+import net.mcreator.ui.validation.component.VTextField;
+
+import javax.annotation.Nullable;
+import javax.swing.*;
+import java.awt.*;
+import java.awt.datatransfer.StringSelection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+public class JStringListField extends JPanel {
+
+	private final List<String> textList = new ArrayList<>();
+	private int index = -1;
+
+	private final JButton edit = new JButton(UIRES.get("16px.edit.gif")) {
+		@Override public String getName() {
+			return "TechnicalButton";
+		}
+	};
+	private final JButton back = new JButton(UIRES.get("previous")) {
+		@Override public String getName() {
+			return "TechnicalButton";
+		}
+	};
+	private final JButton forward = new JButton(UIRES.get("next")) {
+		@Override public String getName() {
+			return "TechnicalButton";
+		}
+	};
+	private boolean joinEntries = false, uniqueEntries = false;
+
+	private final JLabel label = new JLabel() {
+		@Override public String getToolTipText() {
+			return joinEntries || index < 0 ? null : L10N.t("components.string_list.item", index + 1);
+		}
+	};
+
+	public JStringListField(Window parent, @Nullable Function<VTextField, Validator> validator) {
+		super(new FlowLayout(FlowLayout.CENTER, 7, 5));
+		setBackground((Color) UIManager.get("MCreatorLAF.LIGHT_ACCENT"));
+
+		label.setOpaque(false);
+		label.setFont(label.getFont().deriveFont(16f));
+		ToolTipManager.sharedInstance().registerComponent(label);
+
+		back.setMargin(new Insets(0, 0, 0, 0));
+		back.setBorder(BorderFactory.createEmptyBorder(0, 0, 0, 0));

```

Comment:
You don't need to define `0, 0, 0, 0` as parameters. Providing no params already returns an empty border with 0.

Reply:
Eh, ignored these values when copying from item list field class, sorry.